packages/noya-state/src/reducers/application.ts

```diff
@@ -791,16 +798,30 @@ export function reducer(
         const sharedStyle: Sketch.SharedStyle = {
           _class: 'sharedStyle',
           do_objectID: uuid(),
-          name: 'New Layer Style',
-          value: produce(Models.style, (style) => {
-            style.do_objectID = uuid();
-            return style;
-          }),
+          name: name === undefined || name === '' ? 'New Layer Style' : name,
+          value: produce(
+            style === undefined ? Models.style : style,
+            (style) => {
+              style.do_objectID = uuid();
+              return style;
+            },
+          ),
         };
 
         layerStyles.objects.push(sharedStyle);
         state.sketch.document.layerStyles = layerStyles;
-        state.selectedLayerStyleIds = [sharedStyle.do_objectID];
+
+        if (style === undefined) {
```

Comment:
I find this `if / else` logic confusing, since it's not clear from this part of the code why we would do something fairly different if `style === undefined`.

I'd suggest:

```ts
if (currentTab === 'theme') {
  // Update selection
} else {
  // Modify layers
}
```

When I have more time, I want to revisit how we rely on `currentTab` for different behavior, since I think we can come up with something more clear